lib/vehicle_factory.rb

```diff
@@ -0,0 +1,16 @@
+class VehicleFactory
+  def create_vehicles(registrations)
+    vehicles = []
+    registrations.each do |registration|
+      engine = :ev
+      make = registration[:make]
+      model = registration[:model]
+      vin = registration[:vin_1_10]
+      model_year = registration[:model_year]
+
+      vehicle = Vehicle.new(engine: , make: , model:, vin: , model_year: )
+      vehicles << vehicle
+    end 
+    vehicles
+  end 
+end              
```

Comment:
In the loop, you have assigned values to `engine`, `make`, `model`, `vin`, and `model_year`, but you haven't provided values for these attributes when creating a `Vehicle` object. Ensure that you pass the appropriate values to `Vehicle.new` for each attribute.

Also, since we are creating new Vehicle objects in this class, you must also require the Vehicle class at the top of this file.